client/platform/desktop/backend/server.ts

```diff
@@ -64,6 +64,41 @@ apirouter.post('/dataset/:id/meta', async (req, res, next) => {
   }
 });
 
+/* LOAD Attributes */
+apirouter.get('/dataset/:id/attribute', async (req, res, next) => {
```

Comment:
Should probably note somewhere that this is not an ideal way of loading attributes.  By the time this is called, the attributes were loaded already during metadata load, so this loads metadata twice.

If we suppose that, once web is converted, attributes will always be part of the meta object, why not get rid of this method altogether?

Reply:
It's this way because we aren't managing the attributes metadata directly in the loaded metadata of the webapp.  When we make a change is does a round trip to confirm that it was added properly to the metadata file.
Adding/Editing metadata calls a `setAttributes`, then a `getAttributes` right after to update the current list of attributes.  This was a holdover from utilizing the Global attributes in girder.  I could implement a `useAttributes` to modify the attributes local to the client and then await the response from setting to make sure it was done properly.  I think we may get there in the future and then implement the same upsert/delete sort of updating used for the tracks json.  I was trying to keep this somewhat simple to begin with.

I'll add a note to this that in the future this will probably be swapped for local management of metadata while hitting the canary for saving of the trackJSON and metadata at the same time when the user modifies attributes.

Reply:
Makes sense to me, maybe add a comment, but no other change necessary.